ci/test_runner.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+set -euo pipefail
+
+# Conjur test runner configuration environment:
+# KUBE_NODE_TAG:   Docker tag for the Kubernetes node image. This
+#                  determines the Kubernetes version of the KinD cluster.
+#                  Available versions can be found here:
+#                      https://github.com/kubernetes-sigs/kind/releases
+# CLUSTER_NAME:  Name of KinD cluster to be created.
+# MULTI_NODE:    If set to "true", KinD cluster is created with a master
+#                node and 2 worker nodes. Otherwise, a single-node cluster
+#                is created.
+KUBE_NODE_TAG=${KUBE_NODE_TAG:-v1.17.5}
+CLUSTER_NAME=${CLUSTER_NAME:-v1.17.5}
+MULTI_NODE=${MULTI_NODE:-false}
+IPV6_CLUSTER=${IPV6_CLUSTER:-false}
+
+echo "=============================="
+echo "Test Runner Configuration:"
+echo "KUBE_NODE_TAG: $KUBE_NODE_TAG"
+echo "CLUSTER_NAME:  $CLUSTER_NAME"
+echo "MULTI_NODE:    $MULTI_NODE"
+echo "IPV6_CLUSTER:  $IPV6_CLUSTER"
+echo "=============================="
+
+function delete_cluster {
+    echo "=============================="
+    echo "Deleting Kubernetes cluster..."
+    echo "=============================="
+    kind delete cluster --name "${CLUSTER_NAME}"
+}
+
+echo "==================================="
+echo "Creating KinD configuration file..."
+echo "==================================="
+KUBE_CONFIG="$(pwd)/kind_config.yaml"
+cat > $KUBE_CONFIG <<EOT
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+- role: control-plane
+  image: "kindest/node:$KUBE_NODE_TAG"
+EOT
+if [ "$MULTI_NODE" = "true" ]; then
+  cat >> $KUBE_CONFIG <<EOT
+- role: worker
+  image: "kindest/node:$KUBE_NODE_TAG"
+- role: worker
+  image: "kindest/node:$KUBE_NODE_TAG"
+EOT
+fi
+if [ "$IPV6_CLUSTER" = "true" ]; then
+  cat >> $KUBE_CONFIG <<EOT
+networking:
+  ipFamily: ipv6
+EOT
+fi
+
+echo "=============================="
+echo "Creating Kubernetes cluster..."
+echo "=============================="
+kind create cluster \
+    --config "${KUBE_CONFIG}" \
+    --name "${CLUSTER_NAME}"
+trap delete_cluster EXIT
+
+# When KinD creates a cluster, it creates a Kubernetes config that provides
+# local host access to the Kubernetes API server at address:
+#      127.0.0.1:<random-local-port>
+# However, for this test, the KinD cluster has just been created from within
+# this test client container, which happens to be a Docker peer container
+# of the KinD master node container. Therefore, this container needs access
+# to the Kubernetes API server at this address:
+#      <Master Node Container's IP address>:6443
+echo "Fixing up Kubernetes config file..."
+MASTER_IP=$(docker inspect --format='{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' ${CLUSTER_NAME}-control-plane)
+echo "Master Node IP Address: ${MASTER_IP}"
+sed -i "s/^    server:.*/    server: https:\/\/$MASTER_IP:6443/" /root/.kube/config
```

Comment:
not a blocker but this seems pretty fragile

Reply:
Ooh, yes, very fragile. :o
I changed this to use the `kubectl config set clusters.<cluster-name>.server <server-address` command.